File: src/migrations/migration6.ts
import {Kingdom} from '../kingdom/data/kingdom';
import {Migration} from './migration';
import {isNullable} from '../utils';

export class Migration6 extends Migration {
    constructor() {
        super(6);
    }

    override async migrateKingdom(game: Game, kingdom: Kingdom): Promise<void> {
        if (isNullable(kingdom.homebrewActivities)) {
            kingdom.homebrewActivities = [];
        }
    }
}

File: src/jsMain/kotlin/at/posselt/pfrpg2e/migrations/migrations/Migration6.kt
package at.posselt.pfrpg2e.migrations.migrations

import at.posselt.pfrpg2e.kingdom.KingdomData
import com.foundryvtt.core.Game

class Migration6 : Migration(6) {
    override suspend fun migrateKingdom(game: Game, kingdom: KingdomData) {
        @Suppress("SENSELESS_COMPARISON")
        if (kingdom.homebrewActivities == null) {
            kingdom.homebrewActivities = emptyArray()
        }
    }
}